service/test/io/pedestal/http/route_test.clj

```diff
@@ -955,6 +955,19 @@
        "/?:@-._~!$'()*+,;=/?:@-._~!$'()*+,;=="
        {(keyword "/?:@-._~!$'()* ,;") "/?:@-._~!$'()* ,;=="}))
 
+(deftest t-path-params
```

Comment:
Two quick things:
 * We should add 'negative' tests here to ensure that paths that don't require decoding aren't modified in unintended ways
 * We should add another test that exercises this functionality using the routes tress.